### PR TITLE
Refactoring artemis

### PR DESCRIPTION
**Pull request by azime**
time ./launch_artemis.sh

With refactoring
collected 769 items

- real 34m57.471s

- user 1m48.166s

- sys 	0m30.848s

Without refactoring
collected 267 items

- real 25m53.521s

- user 1m13.017s

- sys 0m17.683s

### CODE REVIEW

File: artemis/tests/bibus_test.py [hunk 1]
```diff
@@ -87,20 +88,27 @@ def _pt_ref_call(self, collection, id, depth):
     # def test_coverage(self):
     #     self.api('/')
 
+    @xfail(reason="Diff. links", raises=AssertionError)
```
Comment (TeXitoi) [inline artemis/tests/bibus_test.py]: These tests were active before. Removing them completely is dangerous.

File: artemis/tests/corr_02_test.py [hunk 2]
```diff
@@ -22,3 +22,18 @@ def test_corr_02_01(self):
...
+@set_scenario({"corr-02": {"scenario": "default"}})
+class TestCorr02Default(Corr02, ArtemisTestFixture):
+    pass
+
+@set_scenario({"corr-02": {"scenario": "new_default"}})
+class TestCorr02NewDefault(Corr02, ArtemisTestFixture):
+    pass
+
+
+#@xfail(reason="Unsupported experimental scenario!", raises=AssertionError)
```
Comment (TeXitoi) [inline artemis/tests/corr_02_test.py]: useless comment.

File: artemis/tests/map_test.py [hunk 3]
```diff
@@ -61,3 +61,15 @@ def test_map_06(self):
...
+    pass
+
+@set_scenario({"map": {"scenario": "new_default"}})
+class TestMapNewDefault(Map, ArtemisTestFixture):
+    pass
+
+
+@set_scenario({"map": {"scenario": "experimental"}})
+class TestMapExperimental(Map, ArtemisTestFixture):
+    pass
```
Comment (TeXitoi) [inline artemis/tests/map_test.py]: lack empty line

File: artemis/tests/art_test_01_test.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                     to="stop_area:TS1:SA:12", datetime="20041214T0700")
+
+
+@set_scenario({"test-01": {"scenario": "default"}})
+class TestTadDefault(ArtTest01, ArtemisTestFixture):
+    pass
+
+@set_scenario({"test-01": {"scenario": "new_default"}})
+class TestTadNewDefault(ArtTest01, ArtemisTestFixture):
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
```
Comment (TeXitoi) [inline artemis/tests/art_test_01_test.py]: I have no difference between the default and new_default, why it is xfail?

File: artemis/tests/art_test_01_test.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    def test_art_test_01_02(self):
+        super(TestTadNewDefault, self).test_art_test_01_02()
+
+
+@set_scenario({"test-01": {"scenario": "experimental"}})
+class TestTadExperimental(ArtTest01, ArtemisTestFixture):
+    @xfail(reason="Unsupported experimental scenario!", raises=AssertionError)
```
Comment (TeXitoi) [inline artemis/tests/art_test_01_test.py]: No difference here

File: artemis/test_mechanism.py [hunk 6]
```diff
@@ -151,45 +167,74 @@ def init_fixture(cls, skip_bina):
...
+
+        if skip_bina:
+            logging.getLogger(__name__).warn("skipping binarization")
             return
 
         for data_set in cls.data_sets:
-            fusio_databases_file = utils.new_fusio_files_path(data_set.name)
-            if not os.path.exists(fusio_databases_file):
+            if data_set.name in cls.dataset_binarized:
+                logging.getLogger(__name__).warn("skipping binarization dataset {}".format(data_set))
```
Comment (xlqian) [inline artemis/test_mechanism.py]: "binarization dataset {} has been done, skipping...."

File: artemis/tests/art_test_01_test.py (hunk 4 above)
Comment (TeXitoi) [inline artemis/tests/art_test_01_test.py]: This one is justified.

File: artemis/tests/art_test_01_test.py (hunk 5 above)
Comment (TeXitoi) [inline artemis/tests/art_test_01_test.py]: justified

File: artemis/tests/art_test_02_test.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+                     to="stop_area:TS2:SA:4", datetime="20041214T0700")
+
+
+@set_scenario({"test-02": {"scenario": "default"}})
+class TestArtTest02Default(ArtTest02, ArtemisTestFixture):
+    pass
+
+@set_scenario({"test-02": {"scenario": "new_default"}})
+class TestArtTest02NewDefault(ArtTest02, ArtemisTestFixture):
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
```
Comment (TeXitoi) [inline artemis/tests/art_test_02_test.py]: justified

File: artemis/tests/bibus_test.py [hunk 8]
```diff
@@ -208,3 +222,47 @@ def test_departures(self):
...
+@set_scenario({"bibus": {"scenario": "new_default"}})
+class TestBibusNewDefault(Bibus, ArtemisTestFixture):
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
+    def test_bibus_01(self):
+        super(TestBibusNewDefault, self).test_bibus_01()
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
+    def test_bibus_03(self):
+        super(TestBibusNewDefault, self).test_bibus_03()
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
+    def test_bibus_04(self):
```
Comment (TeXitoi) [inline artemis/tests/bibus_test.py]: justified

File: artemis/tests/bibus_test.py [hunk 9]
```diff
@@ -208,3 +222,47 @@ def test_departures(self):
...
+        super(TestBibusNewDefault, self).test_bibus_04()
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
+    def test_bibus_09(self):
```
Comment (TeXitoi) [inline artemis/tests/bibus_test.py]: cas intéressant : http://jira.canaltp.fr/browse/ITI-203

File: artemis/tests/nb_corr_04_test.py [hunk 10]
```diff
@@ -13,3 +15,23 @@ def test_nb_corr_04_01(self):
...
+
+
+@set_scenario({"nb-corr-04": {"scenario": "default"}})
+class TestNbCorr04Default(NbCorr04, ArtemisTestFixture):
+    pass
+
+@set_scenario({"nb-corr-04": {"scenario": "new_default"}})
+class TestNbCorr04NewDefault(NbCorr04, ArtemisTestFixture):
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
+    def test_nb_corr_04_01(self):
```
Comment (TeXitoi) [inline artemis/tests/nb_corr_04_test.py]: type are not the same: validate

File: artemis/tests/nb_corr_04_test.py [hunk 11]
```diff
@@ -13,3 +15,23 @@ def test_nb_corr_04_01(self):
...
+        super(TestNbCorr04NewDefault, self).test_nb_corr_04_01()
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
+    def test_nb_corr_04_02(self):
```
Comment (TeXitoi) [inline artemis/tests/nb_corr_04_test.py]: type are not the same: validated

File: artemis/tests/nb_corr_04_test.py [hunk 12]
```diff
@@ -13,3 +15,23 @@ def test_nb_corr_04_01(self):
...
+        super(TestNbCorr04NewDefault, self).test_nb_corr_04_02()
+
+
+@xfail(reason="Unsupported experimental scenario!", raises=AssertionError)
+@set_scenario({"nb-corr-04": {"scenario": "experimental"}})
+class TestNbCorr04Experimental(NbCorr04, ArtemisTestFixture):
+    pass
```
Comment (TeXitoi) [inline artemis/tests/nb_corr_04_test.py]: Valid for experimental: as new_default

File: artemis/tests/poitiers_test.py [hunk 13]
```diff
@@ -70,3 +72,27 @@ def test_poitiers_08(self):
...
+
+
+@set_scenario({"poitiers": {"scenario": "default"}})
+class TestPoitiersDefault(Poitiers, ArtemisTestFixture):
+    pass
+
+@set_scenario({"poitiers": {"scenario": "new_default"}})
+class TestPoitiersNewDefault(Poitiers, ArtemisTestFixture):
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
+    def test_poitiers_08(self):
```
Comment (TeXitoi) [inline artemis/tests/poitiers_test.py]: justified

File: artemis/tests/poitiers_test.py [hunk 14]
```diff
@@ -70,3 +72,27 @@ def test_poitiers_08(self):
...
+@set_scenario({"poitiers": {"scenario": "experimental"}})
+class TestPoitiersExperimental(Poitiers, ArtemisTestFixture):
+    @xfail(reason="Unsupported experimental scenario!", raises=AssertionError)
+    def test_poitiers_04(self):
+        super(TestPoitiersExperimental, self).test_poitiers_04()
+    @xfail(reason="Unsupported experimental scenario!", raises=AssertionError)
+    def test_poitiers_05(self):
+        super(TestPoitiersExperimental, self).test_poitiers_05()
+    @xfail(reason="Unsupported experimental scenario!", raises=AssertionError)
+    def test_poitiers_08(self):
```
Comment (TeXitoi) [inline artemis/tests/poitiers_test.py]: justified

File: artemis/tests/sherbrooke_test.py [hunk 15]
```diff
@@ -82,3 +85,79 @@ def test_sherbrooke_16_sp_to_sp_without_walking(self):
...
+
+@set_scenario({"sherbrooke": {"scenario": "default"}})
+class TestSherbrookeDefault(Sherbrooke, ArtemisTestFixture):
+    pass
+
+
+@set_scenario({"sherbrooke": {"scenario": "new_default"}})
+class TestSherbrookeNewDefault(Sherbrooke, ArtemisTestFixture):
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
+    def test_sherbrooke_01(self):
```
Comment (TeXitoi) [inline artemis/tests/sherbrooke_test.py]: different type: validate

File: artemis/tests/sherbrooke_test.py [hunk 16]
```diff
@@ -82,3 +85,79 @@ def test_sherbrooke_16_sp_to_sp_without_walking(self):
...
+    def test_sherbrooke_02(self):
+        super(TestSherbrookeNewDefault, self).test_sherbrooke_02()
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
+    def test_sherbrooke_06(self):
+        super(TestSherbrookeNewDefault, self).test_sherbrooke_06()
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
+    def test_sherbrooke_10(self):
+        super(TestSherbrookeNewDefault, self).test_sherbrooke_10()
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
+    def test_sherbrooke_13(self):
```
Comment (TeXitoi) [inline artemis/tests/sherbrooke_test.py]: Credible alternative: validated

File: artemis/tests/sherbrooke_test.py [hunk 17]
```diff
@@ -82,3 +85,79 @@ def test_sherbrooke_16_sp_to_sp_without_walking(self):
...
+        super(TestSherbrookeNewDefault, self).test_sherbrooke_13()
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
+    def test_sherbrooke_14(self):
```
Comment (TeXitoi) [inline artemis/tests/sherbrooke_test.py]: different type: validated

File: artemis/tests/tcl_test.py [hunk 18]
```diff
@@ -61,3 +63,33 @@ def test_tcl_07(self):
...
+
+
+@set_scenario({"tcl": {"scenario": "default"}})
+class TestTCLDefault(TCL, ArtemisTestFixture):
+    pass
+
+@set_scenario({"tcl": {"scenario": "new_default"}})
+class TestTCLNewDefault(TCL, ArtemisTestFixture):
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
+    def test_tcl_01(self):
```
Comment (TeXitoi) [inline artemis/tests/tcl_test.py]: Another alternative: validated

File: artemis/tests/tcl_test.py [hunk 19]
```diff
@@ -61,3 +63,33 @@ def test_tcl_07(self):
...
+        super(TestTCLNewDefault, self).test_tcl_01()
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
+    def test_tcl_04(self):
```
Comment (TeXitoi) [inline artemis/tests/tcl_test.py]: Another alternative: validated

File: artemis/tests/tcl_test.py [hunk 20]
```diff
@@ -61,3 +63,33 @@ def test_tcl_07(self):
...
+@set_scenario({"tcl": {"scenario": "experimental"}})
+class TestTCLExperimental(TCL, ArtemisTestFixture):
+    @xfail(reason="Unsupported experimental scenario!", raises=AssertionError)
+    def test_tcl_01(self):
+        super(TestTCLExperimental, self).test_tcl_01()
+    @xfail(reason="Unsupported experimental scenario!", raises=AssertionError)
+    def test_tcl_04(self):
+        super(TestTCLExperimental, self).test_tcl_04()
+    @xfail(reason="Unsupported experimental scenario!", raises=AssertionError)
+    def test_tcl_05(self):
```
Comment (TeXitoi) [inline artemis/tests/tcl_test.py]: Other alternatives: validated

File: artemis/tests/tcl_test.py [hunk 21]
```diff
@@ -61,3 +63,33 @@ def test_tcl_07(self):
...
+        super(TestTCLNewDefault, self).test_tcl_04()
+    @xfail(reason="Unsupported new_default scenario!", raises=AssertionError)
+    def test_tcl_05(self):
+        super(TestTCLNewDefault, self).test_tcl_05()
+
+
+@set_scenario({"tcl": {"scenario": "experimental"}})
+class TestTCLExperimental(TCL, ArtemisTestFixture):
+    @xfail(reason="Unsupported experimental scenario!", raises=AssertionError)
+    def test_tcl_01(self):
```
Comment (TeXitoi) [inline artemis/tests/tcl_test.py]: other alternative: validated

**Pull request merged**